xbmc/epg/EpgContainer.cpp

```diff
@@ -239,7 +239,17 @@ void CEpgContainer::LoadFromDB(void)
   unsigned int iCounter(0);
   if (m_database.IsOpen())
   {
-    ShowProgressDialog(false);
+    {
+      /* unlock m_critSection before calling ShowProgressDialog() -
+         this is not legal, but works around a deadlock bug (because
+         ShowProgressDialog() calls functions which lock
+         g_graphicsContext); note that ShowProgressDialog() is
+         sometimes called with m_critSection locked and sometimes
+         without; this is a major bug that must be addressed
+         eventually */
```

Comment:
It will. I am working on a split of render thread from main thread. This should cure quite a few similar issues.